src/features/device-state/routes/fleet-state-get-v3.ts

```diff
@@ -0,0 +1,176 @@
+const { api } = sbvrUtils;
+const { UnauthorizedError } = errors;
+
+type FleetStateV3 = {
+	[uuid: string]: Omit<StateV3[string], 'is_managed_by__device' | 'apps'> & {
+		apps: {
+			[uuid: string]: Omit<
+				StateV3[string]['apps'][string],
+				'is_managed_by__device'
+			>;
+		};
+	};
+};
+
+const fleetExpand: Expand = {
+	application_config_variable: {
+		$select: ['name', 'value'],
+		$orderby: {
+			name: 'asc',
+		},
+	},
+	application_environment_variable: {
+		$select: ['name', 'value'],
+	},
+	should_be_running__release: releaseExpand,
+	service: {
+		$select: ['id', 'service_name'],
+		$expand: {
+			service_environment_variable: {
+				$select: ['name', 'value'],
+			},
+			service_label: {
+				$select: ['label_name', 'value'],
+			},
+			is_built_by__image: {
+				$select: ['id', 'is_stored_at__image_location'],
+			},
+		},
+	},
+};
+
+const stateQuery = _.once(() =>
+	api.resin.prepare<{ uuid: string }>({
+		resource: 'application',
+		id: { uuid: { '@': 'uuid' } },
+		options: {
+			$expand: fleetExpand,
+		},
+	}),
+);
+
+const releaseQuery = _.once(() =>
+	api.resin.prepare<{ commit: string; fleetId: number }>({
+		resource: 'release',
+		options: {
+			$filter: {
+				commit: { '@': 'commit' },
+				belongs_to__application: { '@': 'fleetId' },
+				status: 'success',
+			},
+			...releaseExpand,
+		},
+	}),
+);
+
+const getFleet = async (req: Request, uuid: string) => {
+	const fleet = await readTransaction((tx) =>
+		stateQuery()({ uuid }, undefined, { req, tx }),
+	);
+
+	if (fleet == null) {
+		throw new UnauthorizedError();
+	}
+
+	return fleet;
+};
+
+const getSuccessfulReleaseForFleetAndCommit = async (
+	req: Request,
+	commit: string,
+	fleetId: number,
+) => {
+	const [release] = await readTransaction((tx) =>
+		releaseQuery()({ commit, fleetId }, undefined, { req, tx }),
+	);
+
+	if (release == null) {
+		throw new UnauthorizedError();
+	}
+
+	return release;
+};
+
+const getFleetAppsForState = (
+	fleet: AnyObject,
+	release: AnyObject | undefined,
+	config: Dictionary<string>,
+): FleetStateV3[string]['apps'] => {
+	return {
+		[fleet.uuid]: {
+			id: fleet.id,
+			name: fleet.app_name,
+			is_host: fleet.is_host,
+			class: fleet.is_of__class,
+			...(release != null && {
+				releases: buildAppFromRelease(undefined, fleet, release, config),
+			}),
+		},
+	};
+};
+
+const getFleetStateV3 = async (
+	req: Request,
+	uuid: string,
+	releaseUuid: string | undefined = undefined,
+): Promise<FleetStateV3> => {
+	const fleet = await getFleet(req, uuid);
+	const config = getConfig(undefined, fleet);
+	let release: AnyObject | undefined;
+	if (releaseUuid) {
+		release = await getSuccessfulReleaseForFleetAndCommit(
+			req,
+			releaseUuid,
+			fleet.id,
+		);
+	} else {
+		release = fleet.should_be_running__release[0];
+	}
+	const apps = getFleetAppsForState(fleet, release, config);
+
+	const state: FleetStateV3 = {
+		[uuid]: {
+			name: fleet.app_name,
+			apps,
+			config,
+		},
+	};
+	return state;
```

Comment:
You can simplify a bit by directly returning, and that will mean the type is inferred from the function return type rather than needing to be explicitly typed
```suggestion
	return {
		[uuid]: {
			name: fleet.app_name,
			apps,
			config,
		},
	};
```